Dockerfile

```diff
@@ -1,9 +1,12 @@
-FROM golang:1.22-bullseye AS builder
+FROM golang:1.22-bookworm AS builder
+
+ARG TARGETOS
+ARG TARGETARCH
 
 ENV GO111MODULE=on \
     CGO_ENABLED=0 \
-    GOOS=linux \
-    GOARCH=amd64 \
+    GOOS=${TARGETOS} \
+    GOARCH=${TARGETARCH} \
     GOBIN=/dist
```

Comment:
This means it's using emulation to compile for non-native targets.

Perhaps we should consider using `FROM --platform=$BUILDPLATFORM golang:…` and changing `go install …` to `go build -o <OUTPUT_BIN> …` so that it can use the native build platform and cross-compile (which is much faster and isn't prone to emulation errors).

If so, it may be good to change `FROM alpine:…` below to include `--platform=$TARGETPLATFORM`, just to be explicit and clear about which platform is being used where.